02-wrangling.Rmd

```diff
@@ -256,55 +320,182 @@ lang_no_delimiter <- separate(lang_messy_longer,
+
+### Using `mutate` to modify columns
+
+Notice in the table above the word `<chr>` appears beneath each of the column names. The word under the column name indicates the data type of each column. Here all of our variables are "character" data types. Recall, a character data type is a letter or a number surrounded by quotes. In the previous example in section 
+\@ref(pivot-wider), the `most_at_home` and `most_at_work` variables were `<dbl>` (double) (you can verify this by looking at the tables in the previous sections), which is a numeric data type. This change is due to the delimiter "/" when we read in this messy data set. R read the columns in as character types, and it stayed that way after we separated the columns. 
+
+Here it makes sense for `region`, `category`, and `language` to be stored as a character type. However, if we want to apply any functions that treat the `most_at_home` and `most_at_work` columns as a number (e.g. finding rows that are above a numeric threshold of a column), it won't be possible to do if the variable is stored as a `character`. R has a variety of data types, but here we will use the function `mutate` to convert these two columns to a "numeric" data type. `mutate` is a function that will allow us to create a new variable or transform existing ones in our data set. We specify the data set in the first argument, and in the proceeding arguments, we specify the function we want to apply (`as.numeric`) to which columns (`most_at_home`, `most_at_work`). Then we give the mutated variable a new name. Here we are naming the columns the same names ("most_at_home", "most_at_work"), but you can call these mutated variables anything you'd like. 
+
+`mutate`'s general syntax is detailed in Figure \@ref(fig:img-mutate). 
+
+```{r img-mutate, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Syntax for the mutate function", out.width="1100", fig.retina = 2}
+knitr::include_graphics("img/mutate_function.jpeg")
+```
+
+
+Below we use `mutate` to convert the columns `most_at_home` and `most_at_home`
+to numeric data types in the `official_langs` data set as described in figure 
+\@ref(fig:img-mutate):
+
+```{r}
+official_langs <- mutate(official_langs,
   most_at_home = as.numeric(most_at_home),
   most_at_work = as.numeric(most_at_work)
 )
-lang_no_delimiter
+official_langs
 ```
 Now we see `<dbl>` appears under our columns, `most_at_home` and `most_at_work`, indicating they are double data types (which is one of the sub-types of numeric)!
 
-Is this data now tidy? Well, if we recall the three criteria for tidy data:
+Now when we filter we get what we expect: 
+```{r}
+filter(official_langs, most_at_home > 2669195)
+```
+
+In this case, we see that `filter` returns a data frame with only one row; this indicates that only English in Toronto is reported by more people as their primary language at home than French in Montréal.
+
```

Comment:
Can we add another plain vanilla `mutate` example, where we do some math? Or change the value of the column. I think changing types is one of the more less common examples of mutate... More often I am changing the scale of a variable, or combing columns in some way, etc.